module/zfs/arc.c

```diff
@@ -7138,10 +7146,14 @@ l2arc_write_buffers(spa_t *spa, l2arc_dev_t *dev, uint64_t target_sz)
 	/*
 	 * Copy buffers for L2ARC writing.
 	 */
-	for (try = 0; try <= 3; try++) {
+	for (try = 0; try < L2ARC_FEED_TYPES; try++) {
 		multilist_sublist_t *mls = l2arc_sublist_lock(try);
 		uint64_t passed_sz = 0;
 
+		/* Something unexpected happens. */
+		if (mls == NULL)
+			break;
```

Comment:
The comment isn't necessary here and a `continue` is more appropriate than a `break`.  This allows us to move on to the next `multilist_sublist_t`.  Although in practice none of this can ever happen.